Automattic/pocket-casts-android
# Transcripts - Search logic

## Description

This implements transcript search logic.

Not in scope:
- Scroll to highlighted text
- Text styling Figma match

## Testing Instructions
1. Load transcript view for an episode with transcript (e.g. Cautionary Tales https://pcast.pocketcasts.net/8zpmjiru)
2. ✅ Notice that search icon is visible when transcript is shown
3. Tap on search icon
4. Search a text
5. ✅ Notice that 
    - Search occurrences text is updated in the search bar (if search results are not found, 0 is displayed)
    - Up, down arrow buttons are enabled if search results found
    - Search occurrences are highlighted in the transcript
8. Tap on up arrow in the search bar
9. ✅ Notice that next search result is highlighted (scroll to highlighted text is not added in this PR)
10. Tap on previous arrow in the search bar
11. ✅ Notice that previous search result is highlighted (scroll to highlighted text is not added in this PR) 
12. Tap on x icon in the search bar
13. ✅ Notice that the highlight style is removed
14. Stress test on a large transcript (e.g. for episode https://pca.st/1iofzvym)
15. ✅ Notice that search results are shown without any lag
16. Test search on HTML transcript (e.g. episode from If books could kill https://pca.st/tqib657g)
17. ✅ Notice that search results are shown for HTML transcript 

## Screenshots or Screencast 

https://github.com/user-attachments/assets/88cb5d32-e044-4ee4-99cb-b873c4351912


## Checklist
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics. - Analytics will be added in a separate PR
 
#### I have tested any UI changes...
